### PR TITLE
Fix links verification on ProjectDetail/ProjectsLibrary pages 

### LINKED ISSUE #26: Project cards have links that aren't open for the public
Format for adding @ChialiT - `[Project name]: [Link type], [Link type]`

Example: 
- CryptKeeper: website

### PR DESCRIPTION
**Pull request by samueloh99**
Closes #26 #25 #24 #30

### CODE REVIEW

**Review by ChialiT — COMMENTED** [review]
it seems working, thank you.

File: app/projects/[id]/page.tsx [hunk 1]
```diff
@@ -45,24 +45,17 @@ export default function ProjectDetailPage() {
...
-                    alt="bg"
-                    width={20}
-                    height={20}
-                  />
-                  <p>Github</p>
-                </div>
-              </NextLink>
-            )}
-            {websiteLink !== "none" && (
+            <NextLink
```
Comment (thebeyondr) [inline app/projects/[id]/page.tsx:48]: why did you remove the conditional checking for github?

File: app/projects/[id]/page.tsx [hunk 2]
```diff
@@ -79,7 +72,7 @@ export default function ProjectDetailPage() {
                 </div>
               </NextLink>
             )}
-            {twitterLink !== "none" && (
+            {twitterLink !== null && (
```
Comment (thebeyondr) [inline app/projects/[id]/page.tsx:75]:
```suggestion
            {!!twitterLink && (
```

File: app/projects/page.tsx [hunk 3]
```diff
@@ -73,20 +73,22 @@ export default function ProjectsPage() {
...
-                      >
-                        <Image
-                          src={GlobeIcon}
-                          className="cursor-pointer"
-                          alt="globalVector"
-                          width={18}
-                          height={18}
-                        />
-                      </Link>
+                      {project.links.website !== null && (
```
Comment (thebeyondr) [inline app/projects/page.tsx:76]:
```suggestion
                      {!!project.links.website && (
```

File: config/projects.ts [hunk 4]
```diff
@@ -9,8 +9,8 @@ export const projects = [
     links: {
       github: "https://github.com/Rate-Limiting-Nullifier/circom-rln",
       website: "https://rate-limiting-nullifier.github.io/rln-docs/",
-      twitter: "none",
-      discord: "none",
+      twitter: null,
```
Comment (thebeyondr) [inline config/projects.ts:12]:
```suggestion
      twitter: "",
```

```suggestion
      
```

choose one. having null in the midst of strings is a weird choice. This is also a sign that this file needs to be typed.

File: config/projects.ts
Comment (thebeyondr) [inline config/projects.ts:1]: This file should be typed for consistency. `null` values should be implied or the field removed if they don't exist as per  #24

File: components/svgs/social-medias.tsx
Comment (thebeyondr) [inline components/svgs/social-medias.tsx:1]: `fill` props can be removed if empty. the style of their parent will cascade

File: app/page.tsx [hunk 5]
```diff
@@ -33,7 +33,7 @@ export default function IndexPage() {
             />
           </Link>
         </div>
-        <div className="m-auto flex h-[364px] w-full max-w-[280px] items-center justify-center md:m-0 md:h-full md:w-full lg:max-w-[542px]">
+        <div className="m-auto flex h-[364px] w-full max-w-[280px] items-center justify-center md:m-0 md:h-full md:w-full lg:max-w-[400px]">
           <Image src={PSELogo} alt="pselogo" style={{ objectFit: "cover" }} />
```
Comment (thebeyondr) [inline app/page.tsx:37]:
```suggestion
          <Image src={PSELogo} alt="pselogo" className="object-cover h-{image height in px} w-full"/>
```
Think you need to set the height so the aspect ratio is respected? Not sure why the switch to using `style` here either.

File: components/svgs/arrows.tsx [hunk 6]
```diff
@@ -10,7 +10,7 @@ export const ArrowRightUp = ({ color }: MySvgProps) => {
       width="24"
       height="24"
       viewBox="0 0 24 24"
-      fill="none"
+      fill=""
```
Comment (thebeyondr) [inline components/svgs/arrows.tsx:13]:
```suggestion
   
```
remove the fill prop

**Review by thebeyondr — CHANGES_REQUESTED** [review]
pls

File: app/projects/[id]/page.tsx (hunk 1 above)
Comment (samueloh99) [reply app/projects/[id]/page.tsx:48]: because every project have github

File: app/page.tsx (hunk 5 above)
Comment (samueloh99) [reply app/page.tsx:37]: because now you have to use style on next/image.
and im already setting the w and h on div above

File: config/projects.ts
Comment (samueloh99) [reply config/projects.ts:1]: For me working with null is fine.. I have been working with this till now..
but i will remove the field... if we dont have the link..

File: app/projects/[id]/page.tsx [hunk 7]
```diff
@@ -32,7 +32,6 @@ export default function ProjectDetailPage() {
     (project) => String(project.id) === router.split("/").slice(1)[1]
   )[0]
 
-  const githubLink = findProject.links.github
   const twitterLink = findProject.links.twitter
```
Comment (thebeyondr) [inline app/projects/[id]/page.tsx:35]: Not sure why these variables are here in the first place, but you should be consistent in your usage. Meaning you should be using githubLink, twitterLink, websiteLink from here on out. This is not currently so below. Also `const {twitter: twitterLink, websiteLink, github:githubLink} = findProject.links`  would yield same results in one line

File: app/projects/[id]/page.tsx [hunk 8]
```diff
@@ -45,56 +44,39 @@ export default function ProjectDetailPage() {
...
+            <Link
+              href={findProject.links.github}
+              target="_blank"
+              rel="noreferrer"
+            >
+              <div className="flex items-center gap-2">
+                <Image src={GithubVector} alt="bg" width={20} height={20} />
+                <p>Github</p>
+              </div>
+            </Link>
```
Comment (thebeyondr) [inline app/projects/[id]/page.tsx:56]: This doesn't check if github links exist. Why?

File: app/projects/page.tsx [hunk 9]
```diff
@@ -73,20 +73,22 @@ export default function ProjectsPage() {
                           height={18}
                         />
                       </Link>
```
Comment (thebeyondr) [inline app/projects/page.tsx:75]: No check for github link existence. Not all projects will have Github links

File: components/site-footer.tsx [hunk 10]
```diff
@@ -75,8 +75,20 @@ export function SiteFooter() {
           </NextLink>
         </div>
         <div className="flex gap-5 text-white">
-          <h1>Privacy Policy</h1>
-          <h1>Terms of use</h1>
+          <NextLink
```
Comment (thebeyondr) [inline components/site-footer.tsx:78]: use `Link` instead of `NextLink`, `Image` over  `NextImage`

**Review by thebeyondr — CHANGES_REQUESTED** [review]
Links are only to be shown on the condition that they exist

**Review by thebeyondr — APPROVED** [review]
gonna merge

**Pull request merged**